bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java

```diff
@@ -442,4 +442,61 @@ public void populateDisplayData(DisplayData.Builder builder) {
     builder.add(
         DisplayData.item("readRowsRequest", getDisplayValue(request)).withLabel("ReadRowsRequest"));
   }
+
+  private static class RequestWithScanValueProvider
+      implements ValueProvider<ReadRowsRequest>, Serializable {
+    private final byte[] start;
+    private final byte[] stop;
+    private final int maxVersion;
+    private final Filter filter;
+    private ReadRowsRequest cachedRequest;
+
+    RequestWithScanValueProvider(
+        byte[] start,
+        byte[] stop,
+        int maxVersion,
+        Filter filter
+    ){
+      this.start = start;
+      this.stop = stop;
+      this.maxVersion = maxVersion;
+      this.filter = filter;
+    }
+
+    @Override
+    public ReadRowsRequest get() {
+      if(cachedRequest == null ){
+        Scan scan = new Scan();
+        if (start != null && start.length != 0) {
+          scan.setStartRow(start);
+        }
+        if (stop != null && stop.length != 0) {
+          scan.setStopRow(stop);
+        }
+        if (maxVersion != 0) {
+          scan.setMaxVersions(maxVersion);
+        }
+        if(filter != null){
+          scan.setFilter(filter);
+        }
+        ReadHooks readHooks = new DefaultReadHooks();
+        ReadRowsRequest.Builder builder = Adapters.SCAN_ADAPTER.adapt(scan, readHooks);
+        cachedRequest = readHooks.applyPreSendHook(builder.build());
+      }
+      return cachedRequest;
+    }
+
+    @Override
+    public boolean isAccessible() {
+      return true;
```

Comment:
returning true as all the values will be available, once Query comes in the picture we will check against `tableId`, `projectId` and `instanceId`